ex04.ex

```diff
@@ -75,7 +99,22 @@ defmodule Ex04 do
   return value will be the thing you have to manipulate.
   """
 
-  def even_odd . . . "your code"
+  def even_odd([ h | t ]) do
```

Comment:
### -5

Again, there's a fair amount of unnecessary (and duplicated) logic here. I'd like you to start thinking more in terms of functions and less in terms of procedural programming:

~~~ elixir
  def even_odd(list) do
    list
    |> reverse()
    |> reduce({ [], [] }, &partition_even_odd/2)
  end

  defp partition_even_odd(x, { es, os }) when Integer.is_even(x), do: {[x | es], os}
  defp partition_even_odd(x, { es, os }) when Integer.is_odd(x),  do: {es, [x | os]}
~~~